File: 101/scala/src/test/scala/GreatestCommonDenominatorTest.scala
import org.scalatest.funsuite.AnyFunSuite
import org.scalatest.matchers.should.Matchers

class GreatestCommonDenominatorTest extends AnyFunSuite with Matchers {
  test("Computes greatest common denominator") {
    GreatestCommonDenominator(5, 10) should be(5)
    GreatestCommonDenominator(5, 20) should be(5)
    GreatestCommonDenominator(15, 120) should be(15)
    GreatestCommonDenominator(14, 21) should be(7)
    GreatestCommonDenominator(45, 13) should be(1)
  }
}

File: 101/ts/test/greatestCommonDenominator.test.ts
import {greatestCommonDenominator} from '../src/greatestCommonDenominator';

describe('GreatestCommonDenominator', () => {
  it('should calculate greatest common denominator', () => {
    expect(greatestCommonDenominator(10, 15)).toBe(5);
    expect(greatestCommonDenominator(40, 60)).toBe(20);
  });

  it('should return 1 for no GCD', () => {
    expect(greatestCommonDenominator(13, 45)).toBe(1);
  });
});
